Formula/i/ispc.rb

```diff
@@ -32,13 +32,19 @@ class Ispc < Formula
     depends_on "tbb"
   end
 
-  fails_with gcc: "5"
-
   def llvm
     deps.map(&:to_formula).find { |f| f.name.match? "^llvm" }
   end
 
   def install
+    if OS.linux?
+      inreplace "cmake/CommonStdlibBuiltins.cmake", "foreach (bit 32 64)", "foreach (bit #{Hardware::CPU.bits})"
+      inreplace "cmake/GenerateBuiltins.cmake" do |s|
+        s.gsub! "builtin_to_cpp(32 linux x86)", ""
+        s.gsub! "x86_64-unknown-unknown", "x86_64-unknown-linux-gnu"
```

Comment:
These two targets aren't really the same. Not sure that shipping a bottle that is misconfigured is better than not shipping one at all.

Reply:
`unknown-unknown`'s goal here is to get Clang to pick the default.

Upstream was also considering alternative to hardcoding `x86_64-apple-macosx` to fix macOS x86 build.

Reply:
Why aren't they just using what `clang --print-target-triple` prints?

Reply:
Build should be using Clang as it runs `find_program(CLANGPP_EXECUTABLE NAMES clang++ ...)` which evades our shims due to `CMAKE_PREFIX_PATH`

Referenced cross compilation doc was https://clang.llvm.org/docs/CrossCompilation.html#general-cross-compilation-options-in-clang
>  or you can choose `unknown` and the defaults will be used

---

Though not sure about issue finding glibc headers if target was supposed to use default for triple.

Build included `/usr/include` headers but not the host `/usr/include/x86_64-linux-gnu`.

Reply:
The cross-compilation doc is not quite accurate. Writing `unknown` for the vendor is typical, and doesn't matter, but it has specific meanings for `sys` and/or `env`.

We can see this by comparing the flags used by the clang driver:

<details>
<summary>x86_64-unknown-linux-gnu</summary>

```
Homebrew clang version 19.1.3
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/bin
System configuration file directory: /home/linuxbrew/.linuxbrew/etc/clang
User configuration file directory: /home/linuxbrew/.config/clang
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "/home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/bin/clang-19" -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -dumpdir /dev/null- -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name - -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/tmp -v -fcoverage-compilation-dir=/tmp -resource-dir /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/backward -internal-isystem /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdeprecated-macro -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcxx-exceptions -fexceptions -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/--3ffb8e.o -x c++ -
clang -cc1 version 19.1.3 based upon LLVM 19.1.3 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/backward
 /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
 "/usr/bin/ld" -z relro --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -pie -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o /dev/null /lib/x86_64-linux-gnu/Scrt1.o /lib/x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/11/crtbeginS.o -L/home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/bin/../lib/x86_64-unknown-linux-gnu -L/home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19/lib/x86_64-unknown-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/11 -L/usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/lib -L/usr/lib /tmp/--3ffb8e.o -v -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/11/crtendS.o /lib/x86_64-linux-gnu/crtn.o
```

</details>

<details>
<summary>x86_64-unknown-unknown</summary>

```
Homebrew clang version 19.1.3
Target: x86_64-unknown-unknown
Thread model: posix
InstalledDir: /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/bin
System configuration file directory: /home/linuxbrew/.linuxbrew/etc/clang
User configuration file directory: /home/linuxbrew/.config/clang
 "/home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/bin/clang-19" -cc1 -triple x86_64-unknown-unknown -emit-obj -dumpdir /dev/null- -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name - -mrelocation-model static -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/tmp -v -fcoverage-compilation-dir=/tmp -resource-dir /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19 -fdeprecated-macro -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcxx-exceptions -fexceptions -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/--317737.o -x c++ -
clang -cc1 version 19.1.3 based upon LLVM 19.1.3 default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /home/linuxbrew/.linuxbrew/Cellar/llvm/19.1.3/lib/clang/19/include
 /usr/include
End of search list.
 "/usr/bin/g++" -m64 -o /dev/null /tmp/--317737.o -Wl,-v
collect2 version 11.4.0
/usr/bin/ld -plugin /usr/lib/gcc/x86_64-linux-gnu/11/liblto_plugin.so -plugin-opt=/usr/lib/gcc/x86_64-linux-gnu/11/lto-wrapper -plugin-opt=-fresolution=/tmp/ccQIQ8lf.res -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc -plugin-opt=-pass-through=-lc -plugin-opt=-pass-through=-lgcc_s -plugin-opt=-pass-through=-lgcc --build-id --eh-frame-hdr -m elf_x86_64 --hash-style=gnu --as-needed -dynamic-linker /lib64/ld-linux-x86-64.so.2 -pie -z now -z relro -o /dev/null /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/Scrt1.o /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/11/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/11 -L/usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/11/../../.. /tmp/--317737.o -v -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/11/crtendS.o /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/crtn.o
```

</details>

Passing `x86_64-unknown-unknown` makes `clang` invoke `/usr/bin/gcc` as the linker, for starters. The assembly flags are also different. Haven't checked the link flags carefully, but they seem different (but could be equivalent; unclear).

Reply:
Saying `unknown` picks the default is a little misleading too -- it picks some unspecified default configuration rather than the default triple:
```
❯ echo 'int main(){}' | /usr/bin/clang -xc - --target=arm64-unknown-unknown-unknown -c -o test.o
❯ file test.o
test.o: ELF 64-bit LSB relocatable, ARM aarch64, version 1 (SYSV), not stripped
❯ echo 'int main(){}' | /usr/bin/clang -xc - -c -o test.o
❯ file test.o
test.o: Mach-O 64-bit object arm64
```